### PR TITLE
About feature -  firebase analytics injected and implemented

### PR DESCRIPTION
**Pull request by Sashen943**
- Injected into presenter
- Implemented in view controller
- Tests written for new functionality in presenter
- Removed unused file
- General code clean up of about feature

### CODE REVIEW

File: Showcase-iOS/About/View/AboutViewController.swift [hunk 1]
```diff
@@ -32,44 +24,49 @@ class AboutViewController: UIViewController {
...
 
     @IBAction func websiteTapped(_ sender: Any) {
+        aboutPresenter?.trackSocialMediaButtonTap(with: "website")
         if let websiteUrl = aboutViewModel?.website {
             UIApplication.shared.open(websiteUrl, options: [:], completionHandler: nil)
         }
     }
 
     @IBAction func twitterTapped(_ sender: Any) {
+        aboutPresenter?.trackSocialMediaButtonTap(with: "twitter")
```
Comment (pjwelcome) [inline Showcase-iOS/About/View/AboutViewController.swift]: Can we not make this an String enum and use it through there as this is very hardcoded. We can put all the analytics strings in those enums and then they can be referenced from one place.

**Review by pjwelcome — CHANGES_REQUESTED** [review]

File: Showcase-iOS/About/View/AboutViewController.swift (hunk 1 above)
Comment (Sashen943) [reply Showcase-iOS/About/View/AboutViewController.swift]: Will do so now :)

File: Showcase-iOS/About/View/AboutViewController.swift [hunk 2]
```diff
@@ -32,44 +24,49 @@ class AboutViewController: UIViewController {
...
         if let twitterUrl = aboutViewModel?.twitter {
             UIApplication.shared.open(twitterUrl, options: [:], completionHandler: nil)
         }
     }
 
     @IBAction func facebookTapped(_ sender: Any) {
+        aboutPresenter?.trackSocialMediaButtonTap(with: "facebook")
```
Comment (pjwelcome) [inline Showcase-iOS/About/View/AboutViewController.swift]: this code also seems duplicated, Please can you write a function that takes in the presenter, the enum and the url string and does this UIApplication.shared.open functionality

**Review by pjwelcome — DISMISSED** [review]

File: Showcase-iOS/About/View/AboutViewController.swift (hunk 2 above)
Comment (Sashen943) [reply Showcase-iOS/About/View/AboutViewController.swift]: done

**Review by pjwelcome — APPROVED** [review]

**Pull request merged**